photutils/psf/epsf.py

```diff
@@ -595,6 +595,15 @@ def _smooth_epsf(self, epsf_data):
                  [-0.080816, -0.019592, 0.200816, -0.019592, -0.080816],
                  [+0.041632, -0.080816, 0.078368, -0.080816, +0.041632]])
 
+            if not 3.5 < self.oversampling < 4.5:
+                warnings.warn('A quartic smoothing_kernel was requested, but'
+                              'oversampling is set to {}.  This kernel is '
+                              'only meant for use with an oversampling near 4, '
+                              'so photometry may be compromised. A different '
```

Comment:
The `smoothing_kernel` is used for building the ePSF.  It is true that "photometry may be compromised" (due to an incorrect ePSF), but that is a separate step and thus getting a "photometry warning" when not running a photometry routine may lead to confusion.  Perhaps the warning should instead say something like "the constructed ePSF may be compromised"?